src/plugins/dashboard/public/application/actions/export_csv_action.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Datatable } from 'src/plugins/expressions/public';
+import { FormatFactory } from '../../../../data/common/field_formats/utils';
+import { DataPublicPluginStart, exporters } from '../../../../data/public';
+import { downloadMultipleAs } from '../../../../share/public';
+import { Adapters, IEmbeddable } from '../../../../embeddable/public';
+import { ActionByType } from '../../../../ui_actions/public';
+import { CoreStart } from '../../../../../core/public';
+
+export const ACTION_EXPORT_CSV = 'ACTION_EXPORT_CSV';
+
+export interface Params {
+  core: CoreStart;
+  data: DataPublicPluginStart;
+}
+
+export interface ExportContext {
+  embeddable?: IEmbeddable;
+  // used for testing
+  asString?: boolean;
+}
+
+/**
+ * This is "Export CSV" action which appears in the context
+ * menu of a dashboard panel.
+ */
+export class ExportCSVAction implements ActionByType<typeof ACTION_EXPORT_CSV> {
+  public readonly id = ACTION_EXPORT_CSV;
+
+  public readonly type = ACTION_EXPORT_CSV;
+
+  public readonly order = 5;
+
+  constructor(protected readonly params: Params) {}
+
+  public getIconType() {
+    return 'exportAction';
```

Comment:
I'm wondering if the 'download' icon would better suit this action

Reply:
No strong opinion here.
I think I've picked the same icon used in inspector, but I can change it if you have strong opinion here.